sda/cmd/verify/verify.go

```diff
@@ -204,13 +205,45 @@ func main() {
 				continue
 			}
 
-			hr := bytes.NewReader(header)
-			// Feed everything read from the archive file to archiveFileHash
-			mr := io.MultiReader(hr, io.TeeReader(f, archiveFileHash))
+			// var c4ghr *streaming.Crypt4GHReader
+
+			// for _, key := range archiveKeyList {
+			// 	// Feed data read from the archive file into archiveFileHash
+			// 	mr := io.MultiReader(bytes.NewReader(header), io.TeeReader(f, archiveFileHash))
+
+			// 	c4ghr, err = streaming.NewCrypt4GHReader(mr, *key, nil)
+			// 	if (err == nil) && (c4ghr != nil) {
+			// 		break
+			// 	}
+			// 	log.Warnf("Failed to open C4GH decryptor with key: %v, trying next key. Reason: %s", *key, err.Error())
+			// }
+			// if c4ghr == nil {
+			// 	log.Errorf("Failed to open C4GH decryptor stream with all available key(s)")
+
+			// 	continue
+			// }
+			var key *[32]byte
+			var foundKey bool
+			for _, key = range archiveKeyList {
+				size, err := headers.EncryptedSegmentSize(header, *key)
+				if (err == nil) && (size != 0) {
+					foundKey = true
```

Comment:
```suggestion
			for _, k := range archiveKeyList {
				size, err := headers.EncryptedSegmentSize(header, *k)
				if (err == nil) && (size != 0) {
					key = k
```

Instead of these bool variables use a temporary variable inside the loop instead.

Reply:
fixed in 6a59d828